src/Connectors/PhpRedisConnector.php

```diff
@@ -0,0 +1,279 @@
+<?php
+
+namespace Monospice\LaravelRedisSentinel\Connectors;
+
+use Illuminate\Support\Arr;
+use Illuminate\Redis\Connectors\PhpRedisConnector as LaravelPhpRedisConnector;
+use LogicException;
+use Monospice\LaravelRedisSentinel\Connections\PhpRedisConnection;
+use Monospice\LaravelRedisSentinel\Exceptions\RedisRetryException;
+use Redis;
+use RedisSentinel;
+use RedisException;
+
+/**
+ * Initializes PhpRedis Client instances for Redis Sentinel connections
+ *
+ * @category Package
+ * @package  Monospice\LaravelRedisSentinel
+ * @author   Jeffrey Zant <j.zant@slash2.nl>
+ * @license  See LICENSE file
+ * @link     http://github.com/monospice/laravel-redis-sentinel-drivers
+ */
+class PhpRedisConnector extends LaravelPhpRedisConnector
+{
+    /**
+     * Holds the current sentinel servers.
+     *
+     * @var array
+     */
+    protected $servers;
+
+    /**
+     * The number of times the client attempts to retry a command when it fails
+     * to connect to a Redis instance behind Sentinel.
+     *
+     * @var int
+     */
+    protected $connectorRetryLimit = 20;
+
+    /**
+     * The time in milliseconds to wait before the client retries a failed
+     * command.
+     *
+     * @var int
+     */
+    protected $connectorRetryWait = 1000;
+
+    /**
+     * Configuration options specific to Sentinel connection operation
+     *
+     * Some of the Sentinel configuration options can be entered in this class.
+     * The retry_wait and retry_limit values are passed to the connection.
+     *
+     * @var array
+     */
+    protected $sentinelKeys = [
+        'sentinel_timeout' => null,
+        'retry_wait' => null,
+        'retry_limit' => null,
+        'update_sentinels' => null,
+
+        'sentinel_persistent' => null,
+        'sentinel_read_timeout' => null,
+    ];
+
+    /**
+     * Instantiate the connector and check if the required extension is available.
+     */
+    public function __construct()
+    {
+        if (! extension_loaded('redis')) {
+            throw new LogicException('Please make sure the PHP Redis extension is installed and enabled.');
+        }
+
+        if (! class_exists(RedisSentinel::class)) {
+            throw new LogicException('Please make sure the PHP Redis extension is up to date (5.3.4 or greater).');
+        }
+    }
+
+    /**
+     * Create a new Redis Sentinel connection from the provided configuration
+     * options
+     *
+     * @param array $server  The client configuration for the connection
+     * @param array $options The global client options shared by all Sentinel
+     * connections
+     *
+     * @return PhpRedisConnection The Sentinel connection containing a configured
+     * PhpRedis Client
+     */
+    public function connect(array $servers, array $options = [ ])
+    {
+        // Set the initial Sentinel servers.
+        $this->servers = array_map(function ($server) {
```

Comment:
We should extract the options key before setting the servers or it can result trying to connect to the wrong hostname here: https://github.com/monospice/laravel-redis-sentinel-drivers/pull/36/files#diff-3bf85f59585710cebd9ae7787d8b7ffb7df273ce8ffe2e72053ef96b4bb81ccaR157